### PR TITLE
Add "import filtering rule" for "find usages"

### PR DESCRIPTION
**Pull request by vlad20012**
Used to filter out usages from `use` items

### CODE REVIEW

**Comment by vlad20012** [comment]
See "i" button in the left top corner
Peek 2020-01-20 10-23

File: src/main/kotlin/org/rust/ide/search/RsImportFilteringRule.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.intellij.usages.rules.PsiElementUsage
+import org.rust.lang.core.psi.RsUseSpeck
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.ancestorStrict
+
+class RsImportFilteringRule : ImportFilteringRule() {
+    override fun isVisible(usage: Usage, targets: Array<out UsageTarget>): Boolean {
+        val psi = (usage as? PsiElementUsage)?.element as? RsElement ?: return true
+        val useSpeck = psi.ancestorStrict<RsUseSpeck>()
+        return !(useSpeck != null && useSpeck.alias == null)
```
Comment (mchernyavsky) [inline src/main/kotlin/org/rust/ide/search/RsImportFilteringRule.kt]: Why should we check `useSpeck.alias`? Looks like other implementations (e.g. PyCharm, Kotlin) don't do this.

**Comment by vlad20012** [comment]
bors retry

**Comment by lancelote** [comment]
Note: this button (`i`) is missing in CLion.

**Pull request merged**